solver.rb

```diff
@@ -0,0 +1,32 @@
+require_relative 'exception_handler'
+
+class Solver
+  def initialize
+    @exception = ExceptionHandler.new
+  end
+
+  def factorial(num)
+    @exception.msg('factorial') if num.negative?
+    return num if [1, 0].include?(num)
+
+    num * factorial(num - 1)
+  end
```

Comment:
- [ ] Kindly notice that the factorial of 0 is 1 so it should return 1, not the num.

```ruby
def factorial(num)
    @exception.msg('factorial') if num.negative?
    return 1 if [1, 0].include?(num)

    num * factorial(num - 1)
end
```

Reply:
@ShahierNashaat Thanks. Solved...